crud/src/main/java/com/redhat/lightblue/hooks/HookManager.java

```diff
@@ -173,6 +220,10 @@ public HookManager(HookResolver r, JsonNodeFactory factory) {
      */
     public void clear() {
         queuedHooks.clear();
+
+        queuedHooksSizeB = 0;
+        forRequest = null;
+        warnThresholdBreached = false;
```

Comment:
I don't understand. This method resets the state of this object. Is it tightly coupled to how it expects clients to configure the thresholds? Why not reset to be consistent and decoupled? It's less surprising to the developer. This code is full of enough surprises :-).